fs/sync/sync.go

```diff
@@ -522,13 +522,16 @@ func copyEmptyDirectories(f fs.Fs, entries map[string]fs.DirEntry) error {
 	return nil
 }
 
-func parentDirCheck(entries map[string]fs.DirEntry, entry fs.DirEntry) {
+func (s *syncCopyMove) parentDirCheck(entry fs.DirEntry) {
+	if s.DoMove {
```

Comment:
Can you put a commentabove the if - something like

```
	// If we are moving files then we don't want to remove directories with files in
	// from the source as we are about to move them making the directory empty.
```

It took me a while to work  out how your fix worked!